src/visualizers/VisualizerFactorHistogram.ts

```diff
@@ -0,0 +1,206 @@
+               of the series then the first term of the series will be used
+         **/
+        firstIndex: {
+            value: '' as string | number,
+            forceType: 'integer',
+            displayName: 'First Index',
+            required: false,
+        },
+
+        /** md
+- Number of Terms: The number of terms in the series after the first term.
+                   If this goes past the last term of the sequence it will
+                   show all terms of the sequence after the first index.
+         **/
+        terms: {
+            value: this.terms,
+            forceType: 'integer',
+            displayName: 'Number of Terms',
+            required: true,
+        },
+    }
+
+    checkParameters() {
+        const status = super.checkParameters()
+
+        if (this.params.binSize.value < 1) {
+            status.isValid = false
+            status.errors.push('Bin Size can not be less than 1')
+        }
+
+        return status
+    }
+
+    // Obtain the true first index
+    startIndex(): number {
+        if (
+            typeof this.params.firstIndex.value === 'string'
+            || this.firstIndex < this.seq.first
+        ) {
+            return this.seq.first
+        } else {
+            return this.firstIndex
+        }
+    }
+
+    // Obtain the true number of terms
+    endIndex(): number {
+        return Math.min(this.terms + this.startIndex(), this.seq.last)
+    }
+
+    // Create an array with the number of factor of
+    // each element at the corresponding index of the array
+    factorArray(): number[] {
+        const factorArray = []
+        for (let i = this.startIndex(); i < this.endIndex(); i++) {
+            let counter = 0
+            const factors = this.seq.getFactors(i)
+            if (factors) {
+                for (const [base, power] of factors) {
+                    if (base > 0n) {
+                        counter += Number(power)
+                    } else if (base === 0n) {
+                        counter = 0
+                    }
+                }
+            }
+
+            factorArray[i] = counter
+        }
+        return factorArray
+    }
+
+    // Create an array with the frequency of each number
+    // of factors in the corresponding bins
+    binFactorArray(): number[] {
+        const binFactorArray = []
+        const factorArray = this.factorArray()
+        const largestValue = factorArray.reduce(
+            (a, b) => Math.max(a, b),
+            -Infinity
+        )
+        for (let i = 0; i < largestValue + 1; i++) {
+            binFactorArray.push(0)
+        }
+
+        for (let i = 0; i < factorArray.length; i++) {
+            binFactorArray[Math.trunc(factorArray[i] / this.binSize)]++
+        }
+
+        return binFactorArray
+    }
+
+    // Create a number that represents how
+    // many pixels wide each bin should be
+    binWidth(): number {
+        return (0.95 * this.sketch.width) / this.binFactorArray().length
```

Comment:
It seems like this 0.95 is a ["magic number"](https://en.wikipedia.org/wiki/Magic_number_(programming)). You should create a variable for it, and re-use the variable instead of re-typing 0.95.

Reply:
@Decos14, I saw that you used variables for some other constants, but not 0.95. Is it a coincidence that 0.95 is repeated? If not, I would recommend creating a variable for it.

Reply:
It is somewhat of a coincidence, there is no reason for them to both be the same, it just ended up that way. And I did put comments above it so their purpose is documented.